tests/test-components/components/wasi-key-value/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use helper::{ensure_matches, ensure_ok};
+
+use bindings::wasi::keyvalue::store::{Error, open, KeyResponse};
+
+helper::define_component!(Component);
+
+impl Component {
+    fn main() -> Result<(), String> {
+
+        ensure_matches!(open("forbidden"), Err(Error::AccessDenied));
+
+        let store = ensure_ok!(open("default"));
+
+        // Ensure nothing set in `bar` key
+        ensure_ok!(store.delete("bar"));
+        ensure_matches!(store.exists("bar"), Ok(false));
+        ensure_matches!(store.get("bar"), Ok(None));
+        ensure_matches!(keys(&store.list_keys(None)), Ok(&[]));
+
+        // Set `bar` key
+        ensure_ok!(store.set("bar", b"baz"));
+        ensure_matches!(store.exists("bar"), Ok(true));
+        ensure_matches!(store.get("bar"), Ok(Some(v)) if v == b"baz");
+        ensure_matches!(keys(&store.list_keys(None)), Ok([bar]) if bar == "bar");
+        ensure_matches!(keys(&store.list_keys(Some(0))), Ok([bar]) if bar == "bar");
+
+        // Override `bar` key
+        ensure_ok!(store.set("bar", b"wow"));
+        ensure_matches!(store.exists("bar"), Ok(true));
+        ensure_matches!(store.get("bar"), Ok(Some(wow)) if wow == b"wow");
+        ensure_matches!(keys(&store.list_keys(None)), Ok([bar]) if bar == "bar");
+
+        // Set another key
+        ensure_ok!(store.set("qux", b"yay"));
+        ensure_matches!(keys(&store.list_keys(None)), Ok(c) if c.len() == 2 && c.contains(&"bar".into()) && c.contains(&"qux".into()));
+
+        // Delete everything
+        ensure_ok!(store.delete("bar"));
+        ensure_ok!(store.delete("bar"));
+        ensure_ok!(store.delete("qux"));
+        ensure_matches!(store.exists("bar"), Ok(false));
+        ensure_matches!(store.get("qux"), Ok(None));
+        ensure_matches!(keys(&store.list_keys(None)), Ok(&[]));
```

Comment:
Can we add some atomics and batch tests here, please, even if they're only basic smoke tests for now?  I know you have tests in the implementations, but just something that verifies that the APIs work _from the guest_ even if we can't practically test their atomicity or batchitude.  (Particularly CAS because resources.)